### ISSUE TITLE
Add support for ffmpeg

### ISSUE THREAD

**Issue by boojew** [oznu/docker-homebridge#13]
Would you be willing to add ffmpeg support to the container?

**Comment by oznu** [comment]
This is something I have been thinking about as I am planning to integrate some cameras with my HomeKit setup soon - but don't want the container size to blow out.

What plugin did you want to use? What ffmpeg config is required for your use cases?

**Comment by oznu** [comment]
Rather than including `ffmpeg` I could allow it to be installed if an environment variable is set. This would also cover any other packages that Homebridge plugins may require.

See Pull Request #14.

Until this is merged into master you can test with the `extra-packages` tag (x86_64 only).

```
docker run -e "PACKAGES=ffmpeg" oznu/homebridge:extra-packages
```

This will install ffmpeg [3.2.4](https://github.com/FFmpeg/FFmpeg/releases/tag/n3.2.4) and it's dependencies before Homebridge is started.

```
ffmpeg version 3.2.4 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 6.3.0 (Alpine 6.3.0)
  configuration: 
    --prefix=/usr 
    --enable-avresample
    --enable-avfilter
    --enable-gnutls
    --enable-gpl
    --enable-libmp3lame
    --enable-librtmp
    --enable-libvorbis
    --enable-libvpx
    --enable-libxvid
    --enable-libx264
    --enable-libx265
    --enable-libtheora
    --enable-libv4l2
    --enable-postproc
    --enable-pic
    --enable-pthreads
    --enable-shared
    --enable-x11grab
    --disable-stripping
    --disable-static
    --enable-vaapi
    --enable-libopus
    --disable-debug

  libavutil      55. 34.101 / 55. 34.101
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.101 / 57. 56.101
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libavresample   3.  1.  0 /  3.  1.  0
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
```

**Comment by boojew** [comment]
Amazing! I hacked ffmpeg into my container last night, but I know I did it terribly, so I will 100% try this today or tomorrow.

I tried using homebridge-camera-ffmpeg and I could discover it, but anytime I tried to view the feed, it would crash the homeapp. I dont know if this is because of a misconfig or because of an IOS11 bug. My guess is config.

**Comment by boojew** [comment]
I was able to install this image, but still cant get video. I assume the issue is my config, but still havent quite figured it out.

Also, possibly conicidence, after this new image, one of my garage doors is showing as unsupported - even though they are the same. I've tried readding all the devices and it didnt change anything. Anyways - may be a coincidence. Perhaps the persists db is messed up

**Comment by oznu** [comment]
Thanks for testing. I've merged this as well as #16 which may help if you to do more fine-grained config as the container starts up.

Not sure why your garage door plugin would stop working, this image update did not change the Homebridge or Node.js version. You can reset homebridge by deleting the `accessories`, `persist` and `node_modules` folders in your `/homebridge` which might fix it - but you'll have to setup all your connected accessories and automations again which can be painful.

**Comment by boojew** [comment]
Thanks. Will test
